### DISCUSSION TITLE
Just upgraded to the latest version, boom!

### DISCUSSION THREAD

**Discussion by ObviousInvestor** [wlcrs/huawei_solar#585] [Setting up the integration]
After upgrading to the latest version of the integration it appears that several entities (which I use extensively in automations) have been removed (switch_battery_charge_from_grid, select_battery_working_mode, switch_inverter_inverter_on_off).

I've read through the forums & issues & it appears that these functions have now been moved to services. The only service I can find however is "Huawei Solar: Limit the power fed to the grid to percentage" which is not making any sense to me (how to use it in an automation). Currently I would just automate the on/off charge from grid switch and switch it on and off based on the charge percentage of the battery. Can't see how to do that with the current integration.

I'm relatively new to HA so forgive me if I am asking dumb questions.

**Comment by Roving-Ronin** [comment]
@ObviousInvestor 

I'm running the latest version and whilst I don't have a battery, I can see that the entity for Inverter On/Off is still available.
There was a change of naming format recently that saw some entities names change (see: https://github.com/wlcrs/huawei_solar/discussions/571), are you sure your's aren't still there, under a new name?

Suggest going to the inverter main view in HA (that shows all the sensors, configuration, diagnostic etc) and clicking on them there, to check and confirm the entity names.

**Reply by Roving-Ronin** [reply]
@ObviousInvestor Mark,

Did you get anywhere with this? Just looking at another users system (2 x L1 and 1 x LUNA ESS), in which he did a remove and reinstall with v1.3.1 and notice that the switch.inverter_on_off is still there, BUT the name of it in the Configuration Panel is now 'None'. 
Checking the entity ID and it has now also changed to 'switch.inverter_none', checked the secondary inverter and noticed that it is also the same with entity ID of 'switch.inverter_none'

Looking at my own 2 x M1 inverters that used a prior version of WLCRS when first installed and then have been updated to v1.3.1 and (if I remove the manually added name 'Inverter ON/OFF') they show the default switch name also as 'None', given they are an 'old' install they do have the correct entity ID's of 'switch.inverter_inverter_on_off'  and 'inverter_inverter_on_off_2'.

Thinking that might have something to do with the reformatting of entity naming that occurred in v.1.3.1 and in trying to move from switch.inverter_inverter_on_off' / 'switch.inverter_inverter_on_off_2'  to switch.inverter_on_off' / switch.inverter_on_off_2' somethings gone wrong?

**Reply by Roving-Ronin** [reply]
Converted to an issue https://github.com/wlcrs/huawei_solar/issues/621

**Comment by ObviousInvestor** [comment]
Hi,

I actually ended up just rolling back my HA image to the previous version of the solar setup.

It was working perfectly before I uploaded, and it is now I rolled back so just took the easy way out. Spent a couple of hours trying to figure out what had changed but I'm not really familiar with YAML so I kicked the can down the road. I'll probably try and do the update again when I get some time.

On 17 Feb 2024, at 03:10, Roving Ronin ***@***.***> wrote:

﻿

@ObviousInvestor<https://github.com/ObviousInvestor> Mark,

Did you get anywhere with this? Just looking at another users system (2 x L1 and 1 x LUNA ESS), in which he did a remove and reinstall with v1.3.1 and notice that the switch.inverter_on_off is still there, BUT the name of it in the Configuration Panel is now 'None'.
Checking the entity ID and it has now also changed to 'switch.inverter_none', checked the secondary inverter and noticed that it is also the same with entity ID of 'switch.inverter_none'

Looking at my own 2 x M1 inverters that used a prior version of WLCRS when first installed and then have been updated to v1.3.1 and (if I remove the manually added name 'Inverter ON/OFF') they show the default switch name also as 'None', given they are an 'old' install they do have the correct entity ID's of 'switch.inverter_inverter_on_off' and 'inverter_inverter_on_off_2'.

Thinking that might have something to do with the reformatting of entity naming that occurred in v.1.3.1 and in trying to move from switch.inverter_inverter_on_off' / 'switch.inverter_inverter_on_off_2' to switch.inverter_on_off' / switch.inverter_on_off_2' somethings gone wrong?
